src/smc_sagews/smc_sagews/sage_server.py

```diff
@@ -929,8 +929,18 @@ def execute(self, code, namespace=None, preparse=True, locals=None):
         #code   = sage_parsing.strip_leading_prompts(code)  # broken -- wrong on "def foo(x):\n   print(x)"
         blocks = sage_parsing.divide_into_blocks(code)
 
+        try:
+            import sage.repl.interpreter as sage_repl_interpreter
+        except:
+            log("Error - unable to import sage.repl.interpreter")
+
         for start, stop, block in blocks:
-            if preparse:
+            # if import sage.repl.interpreter fails, sag_repl_interpreter is unreferenced
+            try:
+                do_pp = getattr(sage_repl_interpreter, '_do_preparse', True)
```

Comment:
oh! you don't need to do that try/catch here. `getattr` has **3** arguments here.